iroha_p2p/src/tests/mod.rs

```diff
@@ -0,0 +1,65 @@
+#![allow(clippy::unwrap_used)]
+use iroha_actor::{broker::*, prelude::*};
+use iroha_logger::{config::LoggerConfiguration, info};
+use parity_scale_codec::{Decode, Encode};
+use tokio::time::Duration;
+use ursa::{encryption::symm::prelude::ChaCha20Poly1305, kex::x25519::X25519Sha256};
+
+use super::*;
+use crate::peer::PeerId;
+
+#[derive(iroha_actor::Message, Clone, Debug, Decode, Encode)]
+struct TestMessage(String);
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 4)]
+async fn network_create() {
+    drop(iroha_logger::init(LoggerConfiguration::default()).unwrap());
+    info!("Starting network tests...");
+    let broker = Broker::new();
+    let net = Network::<TestMessage, X25519Sha256, ChaCha20Poly1305>::new(
+        broker.clone(),
+        String::from("127.0.0.1:4433"),
+    )
+    .await
+    .unwrap();
+    drop(net.start().await);
+    tokio::time::sleep(Duration::from_millis(100)).await;
+
+    info!("Connecting to peer...");
+    let peer1 = PeerId {
+        address: String::from("127.0.0.1:4433"),
+        public_key: None,
+    };
+    broker.issue_send(Connect { id: peer1.clone() }).await;
+    tokio::time::sleep(Duration::from_millis(100)).await;
+
+    info!("Posting message...");
+    broker
+        .issue_send(Post {
+            data: TestMessage("Some data to send to peer".to_owned()),
+            id: peer1,
+        })
+        .await;
```

Comment:
Does the peer send a message to itself in this case?

Reply:
No, this is the way we publish a message to `peer1`, that will send it to the other end.